File: frontend/src/types/evaluation.ts
export enum EvaluationStatus {
  PENDING = "pending",
  COMPLETED = "completed",
  FAILED = "failed",
}

export interface EvaluationCriterion {
  criterion_name: string;
  score: number;
  feedback: string;
  improvement_suggestion: string;
}

export interface Evaluation {
  id: string;
  prompt_version_id: string;
  overall_score: number | null;
  status: EvaluationStatus;
  error_message: string | null;
  criteria: EvaluationCriterion[];
  created_at: string;
  completed_at: string | null;
}

export interface EvaluationSummary {
  id: string;
  overall_score: number | null;
  status: EvaluationStatus;
  created_at: string;
}

export interface CriteriaBreakdown {
  criterion_name: string;
  avg_score: number;
}

export interface CommonImprovement {
  criterion_name: string;
  occurrence_count: number;
  avg_score: number;
}

export interface CategoryDashboard {
  category: string;
  avg_score: number;
  evaluation_count: number;
  criteria_breakdown: CriteriaBreakdown[];
  common_improvements: CommonImprovement[];
}

export interface DashboardResponse {
  categories: CategoryDashboard[];
  total_evaluations: number;
}

File: backend/app/schemas/evaluation.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import EvaluationStatus, PromptCategory


# ── Response schemas ─────────────────────────────────────────────

class CriterionResponse(BaseModel):
    criterion_name: str
    score: int
    feedback: str
    improvement_suggestion: str

    model_config = {"from_attributes": True}


class EvaluationSummary(BaseModel):
    id: uuid.UUID
    overall_score: float | None
    status: EvaluationStatus
    created_at: datetime

    model_config = {"from_attributes": True}


class EvaluationResponse(BaseModel):
    id: uuid.UUID
    prompt_version_id: uuid.UUID
    overall_score: float | None
    status: EvaluationStatus
    error_message: str | None = None
    criteria: list[CriterionResponse]
    created_at: datetime
    completed_at: datetime | None = None

    model_config = {"from_attributes": True}


# ── Dashboard schemas ────────────────────────────────────────────

class CriteriaBreakdown(BaseModel):
    criterion_name: str
    avg_score: float


class CommonImprovement(BaseModel):
    criterion_name: str
    occurrence_count: int
    avg_score: float


class CategoryDashboard(BaseModel):
    category: PromptCategory
    avg_score: float
    evaluation_count: int
    criteria_breakdown: list[CriteriaBreakdown]
    common_improvements: list[CommonImprovement]


class DashboardResponse(BaseModel):
    categories: list[CategoryDashboard]
    total_evaluations: int
